ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -604,6 +604,39 @@ func (ipfs *Connector) RepoStat(ctx context.Context) (*api.IPFSRepoStat, error)
 	return &stats, nil
 }
 
+// RepoGC performs a garbage collection sweep on the repo.
+func (ipfs *Connector) RepoGC(ctx context.Context) ([]api.IPFSRepoGC, error) {
+	ctx, span := trace.StartSpan(ctx, "ipfsconn/ipfshttp/RepoGC")
+	defer span.End()
+
+	ctx, cancel := context.WithTimeout(ctx, ipfs.config.IPFSRequestTimeout)
```

Comment:
repo GC is going to need a special timeout because it's an operation that can take hours.

Reply:
We can do something similar to what we are doing for pinTimeout.

Reply:
if you ever run repo GC, you will observe that there is a part where no output is produced. That parts corresponds to walking down all the pinned DAGs to mark them so they are not GC'ed. Only when that set is final, GC starts and the call produces output. The first part, without any output, can take hours.

Reply:
After that output is going to come out very fast very probably, meaning we can just go the simple way and read it.

Reply:
a special timeout setting for is still missing

Reply:
Added a special timeout, `RepoGCTimeout` set at 24 hours by default